src/test_helpers/integration.cljs

```diff
@@ -154,6 +162,41 @@
       (-> (wait-for [:profile.login/messenger-started])
           (.then #(assert-messenger-started))))))
 
+(defn- recover-and-login
+  [seed-phrase]
+  (rf/dispatch [:profile.recover/recover-and-login
+                {:display-name (:name constants/recovery-account)
+                 :seed-phrase  seed-phrase
+                 :password     constants/password
+                 :color        "blue"}]))
+
+(defn- enable-testnet!
+  []
+  (rf/dispatch [:profile.settings/profile-update :test-networks-enabled? true {}])
+  (rf/dispatch [:wallet/initialize]))
+
+(defn- recover-multiaccount!
+  []
+  (p/let [masked-seed-phrase (security/mask-data (:seed-phrase constants/recovery-account))
+          [mnemonic key-uid] (validate-mnemonic masked-seed-phrase)]
+    (rf/dispatch [:onboarding/seed-phrase-validated (security/mask-data mnemonic) key-uid])
+    (rf/dispatch [:pop-to-root :profiles])
+    (rf/dispatch [:profile/profile-selected key-uid])
+    (recover-and-login mnemonic)))
+
+
+(defn setup-recovered-account
+  []
+  (if (messenger-started)
+    (p/resolved ::messenger-started)
+    (do
+      (recover-multiaccount!)
+      (-> (wait-for [:profile.login/messenger-started])
```

Comment:
Mixing Promesa with normal promises. Although this is fine, but if we are migrating to Promesa, we should not use raw promises.

Reply:
Very good point, I will take a look at the rest of the code as well. Thank you!